webapp/src/components/Vendor/decentraland/NFTTopbar/SelectedFilters/SelectedFilters.tsx

```diff
@@ -0,0 +1,118 @@
+import { useState, useMemo } from 'react'
+import { Rarity } from '@dcl/schemas'
+import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+import { useCallback, useEffect } from 'react'
+import { Pill } from './Pill/Pill'
+import { Props } from './SelectedFilters.types'
+import styles from './SelectedFilters.module.css'
+import { getCollectionNameByAddress } from './utils'
+import { Mana } from 'decentraland-ui'
+import classNames from 'classnames'
+
+export const SelectedFilters = ({ browseOptions, isLandSection, onBrowse }: Props)  => {
+  const { rarities, network, onlySmart, contracts, wearableGenders, onlyOnSale, emotePlayMode, minPrice, maxPrice, onlyOnRent } = browseOptions;
+  const [collection, setCollection] = useState<string | undefined>(undefined)
+
+  useEffect(() => {
+    const fetchData = async (contract: string, onlyOnSale: boolean | undefined) => {
+      const collectionName = await getCollectionNameByAddress(contract, onlyOnSale)
+      return collectionName;
+    }
+  
+    if (contracts?.length) {
```

Comment:
```suggestion
    if (contracts?.length && !collectionName) {
```
Should we add the `&& !collectionName` check? Because if we toggle the `onlyOneSale` filter, the `useEffect` will be triggered again and the collection name will be fetched once again, even if we already have the name

Reply:
true! good catch! yeah the alternative you mentioned is perfect 👍